### PR TITLE
[Kotlin] Remove kotlin-reflect dependency when not needed

### PR DESCRIPTION
**Pull request by adamkaplan**
Removes un-unsed kotlin-reflect dependency when the combination of network and serialization library do not require it.
- When not using `ok-http3` or `ok-http4`
- **and**
  - not using `moshi`
  - _or_ using `moshi` with `moshiCodeGen=true`

Original PR #5485

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt [hunk 1]
```diff
@@ -173,12 +173,6 @@ open class ApiClient(val baseUrl: String) {
...
 
     protected inline fun <reified T: Any> parseDateToQueryString(value : T): String {
-        /*
-        .replace("\"", "") converts the json object string to an actual string for the query parameter.
-        The moshi or gson adapter allows a more generic solution instead of trying to use a native
-        formatter. It also easily allows to provide a simple way to define a custom date format pattern
-        inside a gson/moshi adapter.
-        */
-        return Serializer.gson.toJson(value, T::class.java).replace("\"", "")
+        return value.toString()
```
Comment (4brunu) [inline samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt]: Hum, I'm curious on why change was generated.
What OS are you using?

Comment (adamkaplan) [reply samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt]: macOS 10.15.3

**Comment by adamkaplan** [comment]
I don't really understand why CircleCI generates VERSION files with 4.3.0-SNAPSHOT while my local test update following the contributing docs provides VERSION files with 4.2.3-SNAPSHOT from a `master` checkout.

**Comment by adamkaplan** [comment]
I don't really understand why CircleCI generates VERSION files with 4.3.0-SNAPSHOT while my local test update following the contributing docs provides VERSION files with 4.2.3-SNAPSHOT from a `master` checkout.

@4brunu do you know?

File: samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt (hunk 1 above)
Comment (4brunu) [reply samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt]: Can you please run `./mvnw package && sh bin/kotlin-client-all.sh` and commit the changes?

Comment (adamkaplan) [reply samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt]: Done, thank you!

Suggest adding the `./mvn package` step to the [contributing steps](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/contributing.md#testing)

**Comment by 4brunu** [comment]
Now the PR looks much better 👍

**Comment by adamkaplan** [comment]
Some builds seem to have failed in random unrelated tests

**Comment by wing328** [comment]
CircleCI tests (kotlin client tests) passed via https://circleci.com/gh/OpenAPITools/openapi-generator/13537

**Comment by wing328** [comment]
@adamkaplan thanks for the PR, which has been merged into master.

**Comment by wing328** [comment]
@adamkaplan thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Pull request merged**